src/FirmwarePlugin/APM/APMSensorIdDecoder.qml

```diff
@@ -0,0 +1,152 @@
+/****************************************************************************
+ *
+ * (c) 2009-2022 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ *
+ * QGroundControl is licensed according to the terms in the file
+ * COPYING.md in the root of the source code directory.
+ *
+ ****************************************************************************/
+
+import QtQuick 2.3
+import QtQuick.Controls 1.2
+import QtQuick.Controls.Styles 1.4
+import QtQuick.Layouts  1.2
+
+import QGroundControl.FactSystem 1.0
+import QGroundControl.Palette 1.0
+import QGroundControl.Controls 1.0
+
+QGCLabel {
+    property Fact fact: Fact { }
+    text: decode(fact)
+
+
+    property var busTypes: {
+        0: '-',
+        1: 'I2C',
+        2: 'SPI',
+        3: 'UAVCAN',
+        4: 'SITL',
+        5: 'MSP',
+        6: 'EAHRS'
+    }
+
+    property var compassTypes: {
+        0x01: 'HMC5883_OLD',
+        0x07: 'HMC5883',
```

Comment:
Should this be in numeric order to avoid confusion ?

Reply:
this is pretty much a direct copy/paste from https://github.com/ArduPilot/MAVProxy/blob/9bf8b00fbc355650b060e546110877c7898baa81/MAVProxy/modules/lib/mp_util.py#L384

I think it is easier to just keep it the same.